include/zephyr/devicetree.h

```diff
@@ -898,6 +898,17 @@
 #define DT_PROP_BY_IDX(node_id, prop, idx) \
 	DT_CAT5(node_id, _P_, prop, _IDX_, idx)
 
+/**
+ * @brief Get the last element of an array type property
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ *
+ * @return a representation of the last element of the property
+ */
+#define DT_PROP_LAST(node_id, prop) \
+	DT_PROP_BY_IDX(node_id, prop, UTIL_CAT(Z_UTIL_DEC_, DT_PROP_LEN(node_id, prop)))
```

Comment:
isn't this a layer violation? `Z_UTIL_DEC_` macros are impl details for `UTIL_DEC()`, why not use the public macro?

Reply:
It was already spotted by @bjarki-andreasen but I forgot to push the changes. It's updated now.